descartes_samplers/include/descartes_samplers/evaluators/impl/distance_edge_evaluator.hpp

```diff
@@ -85,26 +58,17 @@ bool DistanceEdgeEvaluator<FloatType>::evaluate(const Rung_<FloatType>& from,
                    }
                  });
 
-  // Allocate
-  edges.resize(n_start);
-
-  for (std::size_t i = 0; i < n_start; ++i)
+  FloatType cost = static_cast<FloatType>(0.0);
+  for (std::size_t i = 0; i < delta_thresholds.size(); ++i)
   {
-    const auto* start_vertex = from.data.data() + dof * i;
-    for (std::size_t j = 0; j < n_end; ++j)
-    {
-      const auto* end_vertex = to.data.data() + dof * j;
+    FloatType step = end[i] - start[i];
+    if (std::abs(step) > delta_thresholds[i])
+      return std::make_pair(true, cost);
```

Comment:
If the velocity required to go from `start` to `end` is greater than the allowed threshold, then shouldn't you return false?

Reply:
Good catch.